### DISCUSSION TITLE
PR with merge conflicts create on github

### DISCUSSION THREAD

**Discussion by NicoleGrondinAlayacare** [community/community#22333] [New to GitHub]
<p>Hi,</p>
<p>How does one go about resolving conflicts on a PR created throught the github web ui? I’m familiar on how to resolve commits locally but I’m a bit confused as to how to get the state of the PR on my local. I can’t just pull from the branches in question because no commit has been added to these branches. I can only think of redoing the merge locally then pushing it up to the branch in question, but that completely ignores the open PR.</p>
<p>Thanks!!</p>

**Comment by Wabri** [comment]
<p>I learn how to solve merge conflict with github so for me is more easy :slightly_smiling_face: </p>
<ol>
<li>
<p>You get <strong>conflict on PR</strong>.</p>
</li>
<li>
<p>Click the <strong>“Resolve conflicts”</strong> (on the bottom right):</p>
</li>
</ol>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/7/7d7007acb12177a56e60db123394e298c17f54c9.png" data-download-href="/uploads/short-url/hTFHPqV98cGZbpgbbnVTwFLbVsB.png?dl=1" title="conflict.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/12409541/181070228-b44de4d0-a1b1-44a5-8ba1-5c311ca62ce2.png" alt="conflict.png" data-base62-sha1="hTFHPqV98cGZbpgbbnVTwFLbVsB" width="568" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/7/7d7007acb12177a56e60db123394e298c17f54c9_2_568x500.png, https://aws1.discourse-cdn.com/github/original/2X/7/7d7007acb12177a56e60db123394e298c17f54c9.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/7/7d7007acb12177a56e60db123394e298c17f54c9.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/7/7d7007acb12177a56e60db123394e298c17f54c9_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">conflict.png</span><span class="informations">783×689 75.5 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<ol start="3">
<li>Take the file (or files) where you had the conflict and <strong>delete the unwanted lines</strong> with the editor:</li>
</ol>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/c/c0e1314fdd73434b50a4b94d81cda549989dcf36.png" data-download-href="/uploads/short-url/rwi98dGZcACugxuXjBLZSraVNky.png?dl=1" title="conflict2.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/12409541/181070229-ee1229b7-f89d-4727-969c-045018ea595e.png" alt="conflict2.png" data-base62-sha1="rwi98dGZcACugxuXjBLZSraVNky" width="690" height="183" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/c/c0e1314fdd73434b50a4b94d81cda549989dcf36_2_690x183.png, https://aws1.discourse-cdn.com/github/optimized/2X/c/c0e1314fdd73434b50a4b94d81cda549989dcf36_2_1035x274.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/c/c0e1314fdd73434b50a4b94d81cda549989dcf36_2_1380x366.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/c/c0e1314fdd73434b50a4b94d81cda549989dcf36_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">conflict2.png</span><span class="informations">1883×501 38.7 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<ol start="4">
<li>Once you solved the conflict click on <strong>“Mark as resolved”</strong> (on the right) and then <strong>“commit merge”</strong> (the green button above):</li>
</ol>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/d/d3bd005e32f2759f64ae3184f20180de8903be41.png" data-download-href="/uploads/short-url/ud7G9Pvab3ZBz8SucCdgit7AHux.png?dl=1" title="conflict3.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/12409541/181070230-3e9934fc-4ea5-421f-b382-4ae9eb77c33c.png" alt="conflict3.png" data-base62-sha1="ud7G9Pvab3ZBz8SucCdgit7AHux" width="690" height="151" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/d/d3bd005e32f2759f64ae3184f20180de8903be41_2_690x151.png, https://aws1.discourse-cdn.com/github/optimized/2X/d/d3bd005e32f2759f64ae3184f20180de8903be41_2_1035x226.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/d/d3bd005e32f2759f64ae3184f20180de8903be41_2_1380x302.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/d/d3bd005e32f2759f64ae3184f20180de8903be41_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">conflict3.png</span><span class="informations">1885×414 72 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<ol start="5">
<li>This commit merge the branch “master” (in my case) to the “branch-conflict” and <strong>solved the conflict</strong>:</li>
</ol>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/e/eb596654732f3321a1dfa4ea115977707ca25ede.png" data-download-href="/uploads/short-url/xzZK6lcQcxwrvY55MAVhfspITxk.png?dl=1" title="conflict4.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/12409541/181070232-463a9cf3-533c-48c1-885d-1fae004d0dc4.png" alt="conflict4.png" data-base62-sha1="xzZK6lcQcxwrvY55MAVhfspITxk" width="579" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/e/eb596654732f3321a1dfa4ea115977707ca25ede_2_579x500.png, https://aws1.discourse-cdn.com/github/optimized/2X/e/eb596654732f3321a1dfa4ea115977707ca25ede_2_868x750.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/e/eb596654732f3321a1dfa4ea115977707ca25ede.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/e/eb596654732f3321a1dfa4ea115977707ca25ede_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">conflict4.png</span><span class="informations">1056×911 87.1 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>And this is all…</p>
<p>Hope this helps :slightly_smiling_face:  </p>
<p>-Gabriele-</p>
<blockquote>
<p><strong>Mark helpful posts with Accept as Solution to help other users locate important info. Don’t forget to give Kudos for great content!</strong></p>
</blockquote>

**Comment by NicoleGrondinAlayacare** [comment]
<p>Thanks <a class="mention" href="/wabri">@wabri</a> !! That’s a great step by step :slight_smile: Another hypothetical: what should one do where the merge conflict is too complex so github doesn’t give you the option to resolve it in the web UI? (where the lovely <code>resolve conflict</code> button is disabled)</p>

**Comment by Wabri** [comment, marked as answer]
<p>To solve without the web UI, you need to use the local tools: command line or git client.</p>
<ol>
<li>
<strong>Conflict on PR</strong> :</li>
</ol>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/2/2cb5a331d15775a902694a59ce89f852754bfc22.png" data-download-href="/uploads/short-url/6nwaf2D9esNkmxgymLJ6LG9WGWK.png?dl=1" title="firststep.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/12409541/181070237-c18004a7-905a-45a0-a01d-3824fdc9d61e.png" alt="firststep.png" data-base62-sha1="6nwaf2D9esNkmxgymLJ6LG9WGWK" width="686" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/2/2cb5a331d15775a902694a59ce89f852754bfc22_2_686x500.png, https://aws1.discourse-cdn.com/github/original/2X/2/2cb5a331d15775a902694a59ce89f852754bfc22.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/2/2cb5a331d15775a902694a59ce89f852754bfc22.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/2/2cb5a331d15775a902694a59ce89f852754bfc22_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">firststep.png</span><span class="informations">810×590 45.1 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<ol start="2">
<li>
<p>Locally <strong>try to merge “branchConflict” to “master”</strong> with the merge command (you need to stay in the branch you want to merge):</p>
<p>git merge branchConflict master</p>
</li>
</ol>
<p>(you can do this merge even with a git client like: github desktop or gitkraken) If there are conflicts the output will be something like this:</p>
<pre><code>Auto-merging README.md
CONFLICT (content): Merge conflict in README.md
Automatic merge failed; fix conflicts and then commit the result.
</code></pre>
<p>In my case there are only one conflict (in README.md), if there are more than one conflict the output will be more large. This command will create the same lines of the “Resolve conflicts” in github, something like this:</p>
<pre><code># test

&lt;&lt;&lt;&lt;&lt;&lt;&lt; HEAD
This line will create a new conflict
=======
I try to create conflicts

Hope this can help someone
&gt;&gt;&gt;&gt;&gt;&gt;&gt; master
</code></pre>
<p>You can now edit the files where this lines are. Remember that :</p>
<ul>
<li>&lt;&lt;&lt;&lt;&lt;&lt;&lt; <strong><em>HEAD</em></strong> indicates the start of the conflict of the actual branch in my case <strong>branchConflict</strong> ;</li>
<li>this <strong><em>=======</em></strong> is the line to separate <strong>branchConflict</strong> body with <strong>master</strong> body (in this case);</li>
<li>than will be _ <strong>&gt;&gt;&gt;&gt;&gt;&gt;&gt; master  </strong> _this line indicates the end of the conflict with the <strong>master</strong> branch.</li>
</ul>
<p>NB: in the same file can be more of this lines of conflicts something like this:</p>
<pre><code>_\&lt;\&lt;\&lt;\&lt;\&lt;\&lt;\&lt; HEAD conflictBranch 1 ======= master 1 \&gt;\&gt;\&gt;\&gt;\&gt;\&gt;\&gt; master_&lt;u&gt;&lt;strong&gt;lines not in conflict&lt;/strong&gt;&lt;/u&gt;_\&lt;\&lt;\&lt;\&lt;\&lt;\&lt;\&lt; HEAD conflictBranch 2 ======= master 2 \&gt;\&gt;\&gt;\&gt;\&gt;\&gt;\&gt; master_&lt;u&gt;&lt;strong&gt;lines not in conflict&lt;/strong&gt;&lt;/u&gt;_\&lt;\&lt;\&lt;\&lt;\&lt;\&lt;\&lt; HEAD conflictBranch 3 ======= master 3 \&gt;\&gt;\&gt;\&gt;\&gt;\&gt;\&gt; master_&lt;u&gt;&lt;strong&gt;and so on...&lt;/strong&gt;&lt;/u&gt;
</code></pre>
<ol start="3">
<li>
<p><strong>Solve the conflict</strong> in the same way you can do with web UI, in my case I done this:</p>
<h1>test</h1>
<p>This line will create a new conflict</p>
<p>I try to create conflicts</p>
<p>Hope this can help someone</p>
</li>
</ol>
<p>(simply delete the indicators).</p>
<ol start="4">
<li>
<strong>Add, commit and push</strong> this changes and you will find a green button of “Merge pull request”:</li>
</ol>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/3/3b22b137951d8010e119c581eb62c8b561185775.png" data-download-href="/uploads/short-url/8r8uUuSsBgOlPdWfKEMD3XpFAnH.png?dl=1" title="secondyolo.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/12409541/181070239-379add1f-c8d6-48ed-ba4d-d99ba4a46e0b.png" alt="secondyolo.png" data-base62-sha1="8r8uUuSsBgOlPdWfKEMD3XpFAnH" width="655" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/3/3b22b137951d8010e119c581eb62c8b561185775_2_655x500.png, https://aws1.discourse-cdn.com/github/original/2X/3/3b22b137951d8010e119c581eb62c8b561185775.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/3/3b22b137951d8010e119c581eb62c8b561185775.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/3/3b22b137951d8010e119c581eb62c8b561185775_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">secondyolo.png</span><span class="informations">790×603 49.8 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Ask if you have more problems with conflicts :slightly_smiling_face:</p>
<p>-Gabriele-</p>
<blockquote>
<p><strong>Mark helpful posts with Accept as Solution to help other users locate important info. Don’t forget to give Kudos for great content!</strong></p>
</blockquote>

**Comment by jairunet** [comment]
Hello @NicoleGrondinAlayacare and @Wabri thank you for the information posted above.  I am not sure I understand yet how to resolve the conflicts in the UI, and I really would like to get it I wonder if there is a place with videos of other examples?  I appreciate any additional information, for now, I will continue searching online for documentation and explanations of real case examples.

Thank you for the examples again,
